src/cloudimagedirectory/filter/filter.py

```diff
@@ -50,22 +50,22 @@ def _filter_by_unique_names(data):
         if entry.content is None:
             continue
 
-        # Compare the data entry with the last inserted entry of
-        # the same name. If the new entry is older, do nothing.
-        name = entry.content["name"]
+        # Compare the data entry with the last inserted entry with
+        # the same reference. If the new entry is older, do nothing.
+        ref = entry.filename
```

Comment:
I cant follow 100%, could you help me and explain the details a bit? :)

My understanding is that the filename is always unique, so that we never need to compare it with other image data and never filter something? :see_no_evil: